flow/project.py

```diff
@@ -2716,12 +2964,51 @@ def _verify_group_compatibility(self, groups):
         """Verifies that all selected groups can be submitted together."""
         return all(a.isdisjoint(b) for a in groups for b in groups if a != b)
 
-    def _verify_aggregate_project(self, aggregate):
+    def _verify_aggregate_project(self, jobs):
         """Verifies that all aggregates belongs to the same project."""
-        for job in aggregate:
-            if job not in self:
-                raise ValueError("Job {} is not present "
-                                 "in the project".format(job))
+        if jobs is None:
+            return
+
+        jobs_ids = list(map(self.generate_hashed_aggregates, jobs))
+
+        for id in jobs_ids:
+            # jobs is a list converted from a set so we can
+            # expect that an aggregate id only appear once in
+            # jobs_ids
+            if not self._aggregates_ids.get(id, False):
+                raise LookupError(f"Did not find aggregate having id {id} in the project")
+
+    def _generate_aggregate_from_id(self, id):
+        try:
+            return self._aggregates_ids[id]
+        except KeyError:
+            raise LookupError(f"Did not find aggregate having id {id} in the project")
+
+    def _verify_job_aggregate_via_CLI(self, aggregate, jobs):
+        """Verifies that all aggregates matches with the arguments provided from the
+        ``-j`` option of CLI
+        """
+        if jobs is None:
+            return True
+        elif aggregate not in jobs:
+            # Return False if job/aggregate provided from the ``-j`` CLI option
+            # and that job/aggregate does not match.
+            return False
+
+        return True
+
+    def generate_hashed_aggregates(self, jobs):
+        """"Generate hashed id for an aggregate of jobs
+
+        :param jobs:
+            The signac job handles
+        :type jobs:
+            tuple
+        """
+        blob = json.dumps([job.id for job in jobs], sort_keys=True)
```

Comment:
Why use json at all? This should suffice, no?
```suggestion
        blob = ''.join((job.id for job in jobs))
```